### ISSUE TITLE
Nginx's Main HTTP server does not listen to any port

### ISSUE THREAD

**Issue by lelemka0** [Mailu/Mailu#2851]
### Environment

- [x] docker compose
- [ ] kubernetes
- [ ] docker swarm

### Version

- Version: `2.0`

## Description

The newly installed Mailu cannot be accessed. `ERR_CONNECTION_REFUSED`
I checked nginx.conf and found that nginx is not listening on any port.
```
    # Main HTTP server
    server {
      # Favicon stuff
      root /static;
      # Variables for proxifying
      set $admin admin;
      set $antispam antispam:11334;

      set $webmail webmail;


      set $webdav webdav:5232;

      client_max_body_size 58388608;

      # Listen on HTTP only in kubernetes or behind reverse proxy


      # Only enable HTTPS if TLS is enabled with no error and not on kubernetes


      # Remove headers to prevent duplication and information disclosure
      proxy_hide_header X-XSS-Protection;
      proxy_hide_header X-Powered-By;
      ...
```
I think the key to this problem, the following is part of the front log.
```
mailu-front-1  | An unexpected error occurred:
mailu-front-1  | requests.exceptions.ConnectionError: HTTPSConnectionPool(host='acme-v02.api.letsencrypt.org', port=443): Max retries exceeded with url: /directory (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x72fe5f577160>: Failed to establish a new connection: [Errno -3] Try again'))
mailu-front-1  | Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
```
However, in fact the certificate has been successfully issued. I was able to find a valid certificate at `certs/letsencrypt/archive/mailu-ecdsa`. The letsencrypt.log also proves this.
```
2023-06-06 15:47:00,722:DEBUG:certbot._internal.display.obj:Notifying user: Certificate not yet due for renewal
2023-06-06 15:47:00,724:INFO:certbot._internal.main:Keeping the existing certificate
2023-06-06 15:47:00,725:DEBUG:certbot._internal.display.obj:Notifying user: Certificate not yet due for renewal; no action taken.
```

## Replication Steps

Problem occurs after fresh install.

## Observed behaviour

Nginx can't seem to find the certificate and is not listening on any port.

## Expected behaviour

Nginx finds the certificate and correctly listens to port 443.

## Logs

front container's log:
```
# docker compose logs front
mailu-front-1  | Missing cert or key file, disabling TLS
mailu-front-1  | nginx: [warn] conflicting server name "" on 0.0.0.0:80, ignored
mailu-front-1  | 2023/06/06 15:46:29 [notice] 1#1: using the "epoll" event method
mailu-front-1  | 2023/06/06 15:46:29 [notice] 1#1: nginx/1.22.1
mailu-front-1  | 2023/06/06 15:46:29 [notice] 1#1: OS: Linux 5.10.0-8-amd64
mailu-front-1  | 2023/06/06 15:46:29 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
mailu-front-1  | 2023/06/06 15:46:29 [notice] 1#1: start worker processes
mailu-front-1  | 2023/06/06 15:46:29 [notice] 1#1: start worker process 9
mailu-front-1  | 2023/06/06 15:46:29 [notice] 1#1: start worker process 10
mailu-front-1  | Saving debug log to /var/log/letsencrypt/letsencrypt.log
mailu-front-1  | An unexpected error occurred:
mailu-front-1  | requests.exceptions.ConnectionError: HTTPSConnectionPool(host='acme-v02.api.letsencrypt.org', port=443): Max retries exceeded with url: /directory (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x72fe5f577160>: Failed to establish a new connection: [Errno -3] Try again'))
mailu-front-1  | Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
mailu-front-1  | 115.213.60.37 - - [06/Jun/2023:15:46:52 +0000] "GET / HTTP/1.1" 301 162 "-" "Mozilla/5.0 (Windows NT; Windows NT 10.0; zh-CN) WindowsPowerShell/5.1.19041.2673"
mailu-front-1  | Saving debug log to /var/log/letsencrypt/letsencrypt.log
mailu-front-1  | Certificate not yet due for renewal
mailu-front-1  |
mailu-front-1  | - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
mailu-front-1  | Certificate not yet due for renewal; no action taken.
mailu-front-1  | - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
mailu-front-1  | 192.168.203.1 - - [06/Jun/2023:15:47:19 +0000] "GET / HTTP/1.1" 301 162 "-" "curl/7.74.0"
```
certbot log:
```
# docker exec mailu-front-1 cat /var/log/letsencrypt/letsencrypt.log
2023-06-06 15:46:59,345:DEBUG:certbot._internal.main:certbot version: 1.32.0
2023-06-06 15:46:59,346:DEBUG:certbot._internal.main:Location of certbot entry point: /usr/bin/certbot
2023-06-06 15:46:59,346:DEBUG:certbot._internal.main:Arguments: ['-n', '--agree-tos', '-d', '**domain**', '--expand', '--allow-subset-of-names', '-m', 'admin@**domain**', '--standalone', '--cert-name', 'mailu-ecdsa', '--preferred-challenges', 'http', '--http-01-port', '8008', '--keep-until-expiring', '--allow-subset-of-names', '--key-type', 'ecdsa', '--renew-with-new-domains', '--config-dir', '/certs/letsencrypt', '--post-hook', '/config.py']
2023-06-06 15:46:59,348:DEBUG:certbot._internal.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#manual,PluginEntryPoint#null,PluginEntryPoint#standalone,PluginEntryPoint#webroot)
2023-06-06 15:46:59,530:DEBUG:certbot._internal.log:Root logging level set at 30
2023-06-06 15:46:59,533:DEBUG:certbot._internal.plugins.selection:Requested authenticator standalone and installer None
2023-06-06 15:46:59,538:DEBUG:certbot._internal.plugins.selection:Single candidate plugin: * standalone
Description: Spin up a temporary webserver
Interfaces: Authenticator, Plugin
Entry point: standalone = certbot._internal.plugins.standalone:Authenticator
Initialized: <certbot._internal.plugins.standalone.Authenticator object at 0x72aa9222d2d0>
Prep: True
2023-06-06 15:46:59,539:DEBUG:certbot._internal.plugins.selection:Selected authenticator <certbot._internal.plugins.standalone.Authenticator object at 0x72aa9222d2d0> and installer None
2023-06-06 15:46:59,539:INFO:certbot._internal.plugins.selection:Plugins selected: Authenticator standalone, Installer None
2023-06-06 15:46:59,766:DEBUG:certbot._internal.main:Picked account: <Account(RegistrationResource(body=Registration(key=None, contact=(), agreement=None, status=None, terms_of_service_agreed=None, only_return_existing=None, external_account_binding=None), uri='https://acme-v02.api.letsencrypt.org/acme/acct/1145841987', new_authzr_uri=None, terms_of_service=None), 1b8c4dfa780dac23148d9a1c694dd4f0, Meta(creation_dt=datetime.datetime(2023, 6, 6, 15, 11, 5, tzinfo=<UTC>), creation_host='566c2d3c8855', register_to_eff=None))>
2023-06-06 15:46:59,772:DEBUG:acme.client:Sending GET request to https://acme-v02.api.letsencrypt.org/directory.
2023-06-06 15:46:59,777:DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): acme-v02.api.letsencrypt.org:443
2023-06-06 15:47:00,149:DEBUG:urllib3.connectionpool:https://acme-v02.api.letsencrypt.org:443 "GET /directory HTTP/1.1" 200 752
2023-06-06 15:47:00,153:DEBUG:acme.client:Received response:
HTTP 200
Server: nginx
Date: Tue, 06 Jun 2023 15:47:00 GMT
Content-Type: application/json
Content-Length: 752
Connection: keep-alive
Cache-Control: public, max-age=0, no-cache
X-Frame-Options: DENY
Strict-Transport-Security: max-age=604800

{
  "Ikks75NuHnA": "https://community.letsencrypt.org/t/adding-random-entries-to-the-directory/33417",
  "keyChange": "https://acme-v02.api.letsencrypt.org/acme/key-change",
  "meta": {
    "caaIdentities": [
      "letsencrypt.org"
    ],
    "termsOfService": "https://letsencrypt.org/documents/LE-SA-v1.3-September-21-2022.pdf",
    "website": "https://letsencrypt.org"
  },
  "newAccount": "https://acme-v02.api.letsencrypt.org/acme/new-acct",
  "newNonce": "https://acme-v02.api.letsencrypt.org/acme/new-nonce",
  "newOrder": "https://acme-v02.api.letsencrypt.org/acme/new-order",
  "renewalInfo": "https://acme-v02.api.letsencrypt.org/draft-ietf-acme-ari-01/renewalInfo/",
  "revokeCert": "https://acme-v02.api.letsencrypt.org/acme/revoke-cert"
}
2023-06-06 15:47:00,304:DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): r3.o.lencr.org:80
2023-06-06 15:47:00,697:DEBUG:urllib3.connectionpool:http://r3.o.lencr.org:80 "POST / HTTP/1.1" 200 503
2023-06-06 15:47:00,700:DEBUG:certbot.ocsp:OCSP response for certificate /certs/letsencrypt/archive/mailu-ecdsa/cert1.pem is signed by the certificate's issuer.
2023-06-06 15:47:00,702:DEBUG:certbot.ocsp:OCSP certificate status for /certs/letsencrypt/archive/mailu-ecdsa/cert1.pem is: OCSPCertStatus.GOOD
2023-06-06 15:47:00,722:DEBUG:certbot._internal.display.obj:Notifying user: Certificate not yet due for renewal
2023-06-06 15:47:00,724:INFO:certbot._internal.main:Keeping the existing certificate
2023-06-06 15:47:00,725:DEBUG:certbot._internal.display.obj:Notifying user: Certificate not yet due for renewal; no action taken.
```
/etc/nginx/nginx.conf in front container:
```
# docker exec mailu-front-1 cat /etc/nginx/nginx.conf
# Basic configuration
user nginx;
worker_processes auto;
pcre_jit on;
error_log /dev/stderr notice;
pid /var/run/nginx.pid;
load_module "modules/ngx_mail_module.so";
load_module "modules/ngx_stream_module.so";

events {
    worker_connections 1024;
}

http {
    # Standard HTTP configuration with slight hardening
    include /etc/nginx/mime.types;
    default_type application/octet-stream;
    sendfile on;
    keepalive_timeout 65;
    server_tokens off;
    absolute_redirect off;
    resolver 127.0.0.11 valid=30s;





    # Header maps
    map $http_x_forwarded_proto $proxy_x_forwarded_proto {
      default $http_x_forwarded_proto;
      ''      $scheme;
    }
    map $uri $expires {
      default off;
      ~*\.(ico|css|js|gif|jpeg|jpg|png|woff2?|ttf|otf|svg|tiff|eot|webp)$ 97d;
    }

    map $request_uri $loggable {
      /health 0;
      /auth/email 0;
      default 1;
    }
    access_log /dev/stdout combined if=$loggable;

    # compression
    gzip on;
    gzip_static on;
    gzip_types text/plain text/css application/xml application/javascript
    gzip_min_length 1024;
    # TODO: figure out how to server pre-compressed assets from admin container


    # Enable the proxy for certbot if the flavor is letsencrypt and not on kubernetes
    #
    server {
      # Listen over HTTP
      listen 80;

      listen [::]:80;


      location ^~ /.well-known/acme-challenge/ {
          proxy_pass http://127.0.0.1:8008;
      }

      # redirect to https
      location / {
        return 301 https://$host$request_uri;
      }

    }


    # Main HTTP server
    server {
      # Favicon stuff
      root /static;
      # Variables for proxifying
      set $admin admin;
      set $antispam antispam:11334;

      set $webmail webmail;


      set $webdav webdav:5232;

      client_max_body_size 58388608;

      # Listen on HTTP only in kubernetes or behind reverse proxy


      # Only enable HTTPS if TLS is enabled with no error and not on kubernetes


      # Remove headers to prevent duplication and information disclosure
      proxy_hide_header X-XSS-Protection;
      proxy_hide_header X-Powered-By;

      add_header X-Frame-Options 'SAMEORIGIN';
      add_header X-Content-Type-Options 'nosniff';
      add_header X-Permitted-Cross-Domain-Policies 'none';
      add_header Referrer-Policy 'same-origin';

      # mozilla autoconfiguration
      location ~ ^/(\.well\-known/autoconfig/)?mail/config\-v1\.1\.xml {
        rewrite ^ /internal/autoconfig/mozilla break;
        include /etc/nginx/proxy.conf;
        proxy_pass http://$admin;
      }
      # microsoft autoconfiguration
      location ~* ^/Autodiscover/Autodiscover.json {
        rewrite ^ /internal/autoconfig/microsoft.json break;
        include /etc/nginx/proxy.conf;
        proxy_pass http://$admin;
      }
      location ~* ^/Autodiscover/Autodiscover.xml {
        rewrite ^ /internal/autoconfig/microsoft break;
        include /etc/nginx/proxy.conf;
        proxy_pass http://$admin;
      }
      # apple mobileconfig
      location ~ ^/(apple\.)?mobileconfig {
        rewrite ^ /internal/autoconfig/apple break;
        include /etc/nginx/proxy.conf;
        proxy_pass http://$admin;
      }


      location ^~ /.well-known/acme-challenge/ {
          proxy_pass http://127.0.0.1:8008;
      }


      # If TLS is failing, prevent access to anything except certbot

      location / {
        return 403;
      }




      location /internal {
        internal;

        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Authorization $http_authorization;
        proxy_pass_header Authorization;
        proxy_pass http://$admin;
        proxy_pass_request_body off;
        proxy_set_header Content-Length "";
      }

      location /health {
        return 204;
      }
    }

    # Forwarding authentication server
    server {
      # Variables for proxifying
      set $admin admin;

      listen 127.0.0.1:8000;

      location / {
        proxy_pass http://$admin/internal$request_uri;
      }
    }

    # Healthcheck over localhost, for docker
    server {
      listen 127.0.0.1:10204;
      location /health {
        return 204;
      }
    }

    include /etc/nginx/conf.d/*.conf;
}

stream {
    log_format main '$remote_addr [$time_local] '
                '$protocol $status $bytes_sent $bytes_received '
                '$session_time "$upstream_addr" '
                '"$upstream_bytes_sent" "$upstream_bytes_received" "$upstream_connect_time"';
    access_log /dev/stdout main;

    # managesieve
    server {
      listen 14190;
      resolver 127.0.0.11 valid=30s;

      proxy_connect_timeout 1s;
      proxy_timeout 1m;
      proxy_protocol on;
      proxy_pass imap:4190;
    }
}

mail {
    server_name **domain**;
    auth_http http://127.0.0.1:8000/auth/email;
    proxy_pass_error_message on;
    resolver 127.0.0.11 valid=30s;
    error_log /dev/stderr info;





    # Advertise real capabilities of backends (postfix/dovecot)
    smtp_capabilities PIPELINING "SIZE 50000000" ETRN ENHANCEDSTATUSCODES 8BITMIME DSN;
    pop3_capabilities TOP UIDL RESP-CODES PIPELINING AUTH-RESP-CODE USER;
    imap_capabilities IMAP4 IMAP4rev1 UIDPLUS SASL-IR LOGIN-REFERRALS ID ENABLE IDLE LITERAL+;

    # Default SMTP server for the webmail (no encryption, but authentication)
    server {
      listen 10025;
      protocol smtp;
      smtp_auth plain;
      auth_http_header Auth-Port 10025;
    }

    # Default IMAP server for the webmail (no encryption, but authentication)
    server {
      listen 10143;
      protocol imap;
      smtp_auth plain;
      auth_http_header Auth-Port 10143;
      # ensure we talk HAPROXY protocol to the backends
      proxy_protocol on;
    }

    # SMTP is always enabled, to avoid losing emails when TLS is failing
    server {
      listen 25;

      listen [::]:25;


      protocol smtp;
      smtp_auth none;
      auth_http_header Auth-Port 25;
    }

    # All other protocols are disabled if TLS is failing

```

**Comment by aleleba** [comment]
Have the same issue.

**Comment by sephentos** [comment]
Having the same issue since one of the last 2 or 3 releases.

**Comment by rohankm** [comment]
> Having the same issue since one of the last 2 or 3 releases.

having the same issue any solution?

**Comment by nextgens** [comment]
@rohankm please provide the logs of front and your certbot log file

**Comment by rohankm** [comment]
> @rohankm please provide the logs of front and your certbot log file

Im using 
```
TLS_FLAVOR=cert
```

I can login directly via web interface but none of the ports are open
```
    ports:
      #- "127.0.0.1:80:80"
      #- "127.0.0.1:443:443"
      - "127.0.0.1:110:110"
      - "127.0.0.1:143:143"
      - "127.0.1.1:993:993"
      - "127.0.0.1:995:995"
      - "127.0.0.1:25:25"
      - "127.0.1.1:465:465"
      - "127.0.0.1:587:587"
```

also I'm using a reverse proxy that directly maps to the nginx 443 port

**Comment by nextgens** [comment]
You need both port 443 and port 80 to be redirected for TLS_FLAVOR=cert to work.

**Comment by rohankm** [comment]
> You need both port 443 and port 80 to be redirected for TLS_FLAVOR=cert-letsencrypt to work.
> 
> Other mail-related ports should be bound to your external IP and not 127.0.1.1/127.0.0.1

Hi but `TLS_FLAVOR=cert-letsencrypt` is not an option

**Comment by nextgens** [comment]
Sorry I have misread what you wrote, then the only thing you need to do is put 0.0.0.0 instead of 127.0.0.1 and 127.0.1.1 in your docker-compose.yml

**Comment by rohankm** [comment]
> You need both port 443 and port 80 to be redirected for TLS_FLAVOR=cert-letsencrypt to work.
> 
> Other mail-related ports should be bound to your external IP and not 127.0.1.1/127.0.0.1

also according to this guide https://mailu.io/master/reverse.html
I need to forward the 443 port
```
server {
  listen 443;
  server_name mymailhost.tld;

  # [...] here goes your standard configuration

  location / {
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_pass https://localhost:8443;
  }
}
```

**Comment by rohankm** [comment]
> Sorry I have misread what you wrote, then the only thing you need to do is put 0.0.0.0 instead of 127.0.0.1 and 127.0.1.1 in your docker-compose.yml

when I use 0.0.0.0

I get these logos

```
2023/07/06 13:14:28 [info] 12#12: *3 client login failed: "Authentication credentials invalid" while in http auth state, client: 172.18.0.1, server: 0.0.0.0:465, login: "admin123!"
2023/07/06 13:14:29 [error] 13#13: *4 172.18.0.1 could not be resolved (3: Host not found) while in resolving client address, client: 172.18.0.1, server: 0.0.0.0:465
2023/07/06 13:14:31 [info] 13#13: *4 client login failed: "Authentication credentials invalid" while in http auth state, client: 172.18.0.1, server: 0.0.0.0:465, login: "steamsteam1"
2023/07/06 13:14:39 [info] 11#11: *1 client login failed: "Authentication credentials invalid" while in http auth state, client: 172.18.0.1, server: 0.0.0.0:465, login: "appreciable001"
2023/07/06 13:14:43 [info] 14#14: *19 client 172.18.0.1:49750 connected to 0.0.0.0:465
2023/07/06 13:14:45 [error] 14#14: *19 172.18.0.1 could not be resolved (3: Host not found) while in resolving client address, client: 172.18.0.1, server: 0.0.0.0:465
2023/07/06 13:14:49 [info] 15#15: *21 client 172.18.0.1:39622 connected to 0.0.0.0:465
2023/07/06 13:14:51 [error] 15#15: *21 172.18.0.1 could not be resolved (3: Host not found) while in resolving client address, client: 172.18.0.1, server: 0.0.0.0:465
2023/07/06 13:14:55 [info] 14#14: *19 client login failed: "Authentication credentials invalid" while in http auth state, client: 172.18.0.1, server: 0.0.0.0:465, login: "entity@123!"
2023/07/06 13:14:56 [info] 15#15: *21 client login failed: "Authentication credentials invalid" while in http auth state, client: 172.18.0.1, server: 0.0.0.0:465, login: "confine@123"
```

**Comment by rohankm** [comment]
i have configured as you adviced

```
# Header to take the real ip from
REAL_IP_HEADER=X-Real-IP

# IPs for nginx set_real_ip_from (CIDR list separated by commas)
REAL_IP_FROM=172.18.0.0/24
```

but still I'm getting the same logs

**Comment by nextgens** [comment]
@rohankm let's discuss your case on the other ticket you have opened.

This ticket is related to someone using 2.0 (you appear to be running master) and running into a letsencrypt related issue (and you have not configured letsencrypt).

Post your full configuration (including the reverse proxy's) there.

**Comment by vahidalvandi** [comment]
> 172.18.0.0/24

`REAL_IP_FROM=172.18.0.0/24  `

Shouldn't the server's IP be fixed? For example, 185.21.51.6
